src/packet/public_key.js

```diff
@@ -115,8 +115,11 @@ class PublicKeyPacket {
       pos += 4;
 
       // - A one-octet number denoting the public-key algorithm of this key.
-      this.algorithm = enums.read(enums.publicKey, bytes[pos++]);
-      const algo = enums.write(enums.publicKey, this.algorithm);
+      try {
+        this.algorithm = enums.write(enums.publicKey, bytes[pos++]);
```

Comment:
Here it's a bit of a special case, since knowing the public key algorithm is required for parsing it. But I would let `parsePublicKeyParams` throw in that case, as it already does. (Even though I know the old code also already kind of did this.)